hare/eligibility/oracle.go

```diff
@@ -55,61 +61,72 @@ type Oracle struct {
 	activesCache       addGet
 	genesisTotalWeight uint64
 	genesisMinerWeight uint64
-	blocksProvider     goodBlocksProvider
 	cfg                eCfg.Config
 	log.Log
 }
 
-// Returns the relative layer id that w.h.p. we have agreement on its contextually valid blocks
-// safe layer is defined to be the confidence param layers prior to the provided Layer
-func safeLayer(layer types.LayerID, safetyParam types.LayerID) types.LayerID {
-	if layer <= types.GetEffectiveGenesis()+safetyParam { // assuming genesis is zero
-		return types.GetEffectiveGenesis()
-	}
-
-	return layer - safetyParam
-}
-
-func roundedSafeLayer(layer types.LayerID, safetyParam types.LayerID,
-	layersPerEpoch uint16, epochOffset types.LayerID) types.LayerID {
-
-	sl := safeLayer(layer, safetyParam)
-	if sl == types.GetEffectiveGenesis() {
-		return types.GetEffectiveGenesis()
-	}
-
-	se := types.LayerID(sl.GetEpoch()) // the safe epoch
-
-	roundedLayer := se*types.LayerID(layersPerEpoch) + epochOffset
-	if sl >= roundedLayer { // the safe layer is after the rounding threshold
-		return roundedLayer // round to threshold
+// Returns a range of safe layers that should be used to construct the Hare active set for a given target layer
+// Safe layer is a layer prior to the input layer on which w.h.p. we have agreement (i.e., on its contextually valid
+// blocks), defined to be confidence param number of layers prior to the input layer.
+func safeLayerRange(
+	targetLayer types.LayerID,
+	safetyParam types.LayerID,
+	layersPerEpoch types.LayerID,
+	epochOffset types.LayerID,
```

Comment:
I consider `LayerID` to be analogous to Go's datetime module's `Time` and `uint16` to be analogous to `Duration`.

For that reason I think it makes little sense to make these arguments of type `LayerID`.

Reply:
I changed them to uint16. The types we use to represent layers and layer diffs in configs and throughout the code are still messy and should still be cleaned up. Hopefully we can do that in #2489. But, you're right, we shouldn't make it worse here.